go.mod

```diff
@@ -23,29 +26,84 @@ require (
 	github.com/gofrs/uuid v3.2.0+incompatible
 	github.com/gogo/googleapis v1.1.0 // indirect
 	github.com/gogo/protobuf v1.2.1
-	github.com/golang/protobuf v1.2.0
+	github.com/golang/protobuf v1.3.1
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
 	github.com/hashicorp/go-multierror v1.0.0
 	github.com/mdlayher/vsock v0.0.0-20190329173812-a92c53d5dcab
 	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
 	github.com/opencontainers/image-spec v1.0.1 // indirect
 	github.com/opencontainers/runc v1.0.0-rc8
-	github.com/opencontainers/runtime-spec v0.1.2-0.20181106065543-31e0d16c1cb7
+	github.com/opencontainers/runtime-spec v1.0.2-0.20190207185410-29686dbc5559
+	github.com/opencontainers/selinux v1.3.0 // indirect
 	github.com/pkg/errors v0.8.1
-	github.com/prometheus/client_golang v0.9.2 // indirect
+	github.com/seccomp/libseccomp-golang v0.9.1 // indirect
 	github.com/shirou/gopsutil v2.18.12+incompatible
 	github.com/shirou/w32 v0.0.0-20160930032740-bb4de0191aa4 // indirect
 	github.com/sirupsen/logrus v1.4.1
-	github.com/stretchr/testify v1.2.2
+	github.com/stretchr/testify v1.3.0
 	github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2 // indirect
+	github.com/tchap/go-patricia v2.3.0+incompatible // indirect
 	github.com/urfave/cli v1.20.0 // indirect
 	go.etcd.io/bbolt v1.3.1-etcd.8
-	golang.org/x/sync v0.0.0-20181108010431-42b317875d0f
-	golang.org/x/sys v0.0.0-20190507160741-ecd444e8653b
+	golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
+	golang.org/x/sys v0.0.0-20190616124812-15dcb6c0061f
 	google.golang.org/genproto v0.0.0-20181109154231-b5d43981345b // indirect
 	google.golang.org/grpc v1.21.0
-	gotest.tools v2.2.0+incompatible // indirect
+	k8s.io/klog v0.4.0 // indirect
+	k8s.io/kubernetes v1.15.3 // indirect
+	k8s.io/utils v0.0.0-20190829053155-3a4a5477acf8 // indirect
 )
 
 // Workaround for github.com/containerd/containerd issue #3031
 replace github.com/docker/distribution v2.7.1+incompatible => github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible
+
+replace k8s.io/api => k8s.io/api v0.0.0-20190620084959-7cf5895f2711
```

Comment:
Just to make sure I'm understanding correctly, we need these replace directives because they are indirect dependencies and we thus can't specify specific versions in our go.mod? Kind of unfortunate if so and seems like a major deficiency in go modules, but if there's no better way than it is what it is.

May be worth including a comment in the go.mod file explaining this and that they come from the update-gomod.sh script

Reply:
The reality is a bit sad than that. We cannot even build our binaries without having a bunch of replace directives due to https://github.com/kubernetes/kubernetes/issues/79384. 

From the issue above;

> k8s.io/kubernetes is not primarily intended to be consumed as a module. Only the published subcomponents are (and go get works properly with those).

Ideally speaking, the CRI plugin should depend on the published subcomponents, but it may not be possible right now. I haven't looked how hard it would be though.